app/AccountancyModule/Components/Participants/EditParticipantDialog.php

```diff
@@ -46,15 +47,15 @@ protected function createComponentForm(): BaseForm
         Assertion::keyExists($this->participants, $this->participantId);
 
         $participant = $this->participants[$this->participantId];
-        assert($participant instanceof Participant);
+        assert($participant instanceof ParticipatingPerson);
 
         $form = new BaseForm();
 
         if ($this->isAllowedDaysUpdate) {
             $form->addInteger('days', 'Počet dní')
                 ->setRequired('Musíte vyplnit počet dní')
                 ->addRule(BaseForm::MIN, 'Minimální počet dní je %d', 0)
-                ->setDefaultValue($participant->getDays());
+                ->setDefaultValue(method_exists($participant, 'getDays') ? $participant->getDays() : 0);
```

Comment:
Proč nemít `getDays` přímo součástí ParticipatingPerson? at to neresime pres `method_exists` na více místech a logika bude na jednom miste. Jen bych vracel int|null a taky si ten null zmenil na 0 jako default.

Reply:
Jo, no ono se tohle vlastně nemá nikdy volat - `isAllowedDaysUpdate` se pro instruktory nepoužívá a to `method_exists` tu bylo jen pro umlčení statický analýzy. Ale asi větší smysl než přidávání metody mi přijde sem dát assert... Holt tady to ta superclass trochu komplikuje

Reply:
hmm, tak asi assert

Reply:
Ok, tak ten jsem tam dal ;)